packages/jam/node/reader.ts

```diff
@@ -41,22 +43,53 @@ export const importBlocks = async (node: NodeApi, blocksToImport: string[]) => {
   return await node.close();
 };
 
-export function* startBlocksReader(options: BlocksImporterConfig) {
+export function* startBlocksReader(options: BlocksImporterConfig, logger: Logger) {
   const { chainSpec } = options;
   for (const file of options.files) {
     const isJsonFile = isJson(file);
 
-    const block = isJsonFile ? readJsonBlock(file, chainSpec) : readCodecBlock(file, chainSpec);
-    yield block;
+    logger.log`📖 Reading ${file}...`;
+
+    if (isJsonFile) {
+      yield readJsonBlock(file, chainSpec);
+    } else {
+      yield* readCodecBlocks(file, chainSpec);
+    }
   }
 }
 
 const isJson = (f: string) => f.endsWith(".json");
 
-function readCodecBlock(file: string, chainSpec: ChainSpec): BlockView {
-  const codecData = fs.readFileSync(file);
-  const bytes = BytesBlob.blobFrom(new Uint8Array(codecData));
-  return Decoder.decodeObject(Block.Codec.View, bytes, chainSpec);
+function* readCodecBlocks(file: string, chainSpec: ChainSpec): Generator<BlockView> {
+  const fileDescriptor = fs.openSync(file, "r");
+  const bufferSize = 1024 * 1024; // should be at least max block length
+  const buffer = new Uint8Array(bufferSize);
+  let offset = 0;
+  let bytesRead = 0;
+  const getNextChunk = () => {
+    try {
+      bytesRead = fs.readSync(fileDescriptor, buffer, offset, bufferSize, null);
+    } catch (_) {
+      return false;
+    }
+    offset += bufferSize;
+    return true;
+  };
+  while (getNextChunk()) {
+    const decoder = Decoder.fromBlob(buffer.slice(0, bytesRead), chainSpec);
```

Comment:
```suggestion
    const decoder = Decoder.fromBlob(buffer.subarray(0, bytesRead), chainSpec);
```

Reply:
no longer relevant, after we now create a new buffer for every block decoded.